readable-regexes/src/pydis_jam23/web/__init__.py

```diff
@@ -0,0 +1,122 @@
+import io
+import webbrowser
+from typing import Any
+
+from flask import Flask, flash, render_template, request, send_file
+from PIL import Image
+
+from pydis_jam23.codecs import CODECS, Codec, CodecError, CodecParam
+
+app = Flask(__name__)
+app.secret_key = "secret"  # We're only running locally.
+
+# Yep, it's a global mutable variable. It *should* work fine though since
+# there's only one user.
+current_image: Image.Image | None = None
```

Comment:
Flask provides a `g object` to specifically help deal with this. You can read about it [here](https://flask.palletsprojects.com/en/2.3.x/appcontext/#storing-data).
It's essentially a better way to handle dynamic data that need to stay for the lifetime of the app.

Better yet, I would use a server-side Flask session so that you can support more than a single user. The [Flask-Session](https://flask-session.readthedocs.io/en/latest/) extension is what you would use, usually in combination with something like redis.